src/Test/Html/Selector/Internal.elm

```diff
@@ -49,18 +49,51 @@ queryAll selectors list =
 
         selector :: rest ->
             query ElmHtml.Query.query queryAll selector list
-                |> queryAllChildren rest
+                |> queryAll rest
 
 
-queryAllChildren : List Selector -> List ElmHtml -> List ElmHtml
-queryAllChildren selectors list =
-    case selectors of
-        [] ->
-            list
+queryOnlyChildren : List Selector -> List ElmHtml -> List ElmHtml
+queryOnlyChildren selectors list =
+    let
+        querySelectors =
+            List.map toQuerySelector selectors
 
-        selector :: rest ->
-            query ElmHtml.Query.queryChildren queryAllChildren selector list
-                |> queryAllChildren rest
+        parentNode =
```

Comment:
@rtfeldman, not sure if this can be done recursively as you recommended & still keep track of who the parent is?